src/modes/extra/TaikoDrums.cpp

```diff
@@ -0,0 +1,134 @@
+*    Right | DPAD Right      | Left                  |  LEFT RED
+*    L     | ZL              | Left                  |  LEFT BLUE
+*    Up    | ZR              | Right                 |  RIGHT BLUE
+*    A     | A               | A                     |  RIGHT BLUE
+*    B     | B               | B                     |  RIGHT RED
+*    X     | Y               | Y                     |  RIGHT RED
+*    Z     | R               | R                     |  RIGHT BLUE
+*    Y     | X               | Random Song           |  Right Blue (Unintended)
+*    LS    | MINUS / SELECT  | Option                |  N/A
+*    MS    | HOME            | HOME                  |  N/A
+*
+=============IN SONG================ (B)lue, (R)ed
+                        X minus
+                      /     home 
+      B R               R B
+    B     R           R     B
+              start   /
+          B          /B/
+           /         /
+=============IN MENU================
+                            X minus
+                          /     home 
+      Left Down             Y R
+    Left     Left         B     Right
+                start     /
+             Up          /A/
+              /         /
+=============SWITCH CONTROLS=========
+                           X minus
+                         /     home 
+      Left Down            Y R
+    ZL      Right        B     ZR
+                start    /
+            Up          /A/
+             /          /
+=====================================
+**/
+
+#include "modes/extra/TaikoDrums.hpp"
+
+#define ANALOG_STICK_MIN 1
+#define ANALOG_STICK_NEUTRAL 128
+#define ANALOG_STICK_MAX 255
+
+TaikoDrums::TaikoDrums(socd::SocdType socd_type) : ControllerMode(socd_type) {
+    _socd_pair_count = 2;
+    _socd_pairs = new socd::SocdPair[_socd_pair_count]{
+        socd::SocdPair{&InputState::left,    &InputState::right  },
+        socd::SocdPair{ &InputState::down,   &InputState::up     },
+    };
+}
+
+void TaikoDrums::UpdateDigitalOutputs(InputState &inputs, OutputState &outputs) {
+    // LEFT BLUE
+    // INPUTS: L, LEFT, MOD_X
+    // OUTPUTS: ( DPAD LEFT / DPAD UP / ZL / L )
+    outputs.triggerLDigital = inputs.l;
+    outputs.dpadLeft = inputs.left;
+    outputs.dpadUp = inputs.mod_x;
+    // LEFT RED
+    // INPUTS: DOWN, RIGHT, MOD_Y
+    // OUTPUTS: ( DPAD DOWN / DPAD RIGHT ) 
+    outputs.dpadDown = inputs.down; 
+    outputs.dpadRight = inputs.right;
+    //outputs.dpadDown = inputs.mod_y;
+    // RIGHT BLUE
+    // INPUTS: A, Z, UP
+    // OUTPUTS: ( A / X / ZR / R )
+    outputs.a = inputs.a; 
+    outputs.buttonR = inputs.z;
+    outputs.triggerRDigital = inputs.up;
+    // RIGHT RED
+    // INPUTS: C_DOWN, B, X
+    // OUTPUTS: ( B / Y) 
+    //outputs.b = inputs.c_down; 
+    outputs.b = inputs.b;
+    outputs.y = inputs.x;
+    // Control Row (Right Hand Top Row)
+    // INPUTS: R, Y, LS, MS
+    // OUTPUTS: ( A / X / SELECT / HOME )
+    //outputs.a = inputs.r;

```

Comment:
I see a couple instances like this where you have commented out certain inputs but the comments above don't reflect that input not actually being mapped like that. Is there a particular reason why these lines of code/comments are left like this? Other than that, looks like a nice well documented gamemode 😄